### DISCUSSION TITLE
What is the benefit of the yarn.lock feature? (--yarn)

### DISCUSSION THREAD

**Discussion by silvenon** [oven-sh/bun#5982] [Q&A] [answered]
I wasn't able to gather from the documentation, but why does bun support generating `yarn.lock`? Only for inspecting purposes or are there actual use cases for this feature?

**Comment by Bellisario** [comment, marked as answer]
The benefits of using  the `yarn.lock` feature are:

- ability to visually inspect the dependencies lockfile and actually track the file updates with `git`, not possible using the default Bun binary lockfile
- providing a fallback lockfile for Yarn users without Bun installed and getting a deterministic build both for Bun and Yarn   
- using Bun as a Yarn "replacement": installing dependencies with `bun install --yarn --no-save` will prevent Bun from creating a `bun.lockb` in addition to `yarn.lock`, letting you switch to Bun without adding an additional file to your project root

Anyway, if you are not planning to keep deterministic builds for Yarn and you don't need to manually track lockfile changes with `git`, you could consider not keeping an additional lockfile around and just run `bun bun.lockb` to manually see the file content formatted as `yarn.lock` when you need it.

> **Note**\
> From my personal tests, if you don't use the `--yarn` flag each time you run `bun install`, Bun won't track the changes to `yarn.lock`, but only `bun.lockb`, so make sure to remember to use this flag each time, otherwise you will lose the deterministic build for Yarn.

**Reply by silvenon** [reply]
> ability to visually inspect the dependencies lockfile and actually track the file updates with git, not possible using the default Bun binary lockfile

It's possible to inspect `bun.lockb` with the additional `.gitattributes` and git config modifications described in the docs. But that probably only works locally, and only on the machine of the person that configured their local or global config.

> providing a fallback lockfile for Yarn users without Bun installed and getting a deterministic build both for Bun and Yarn

Ah, I see… seems like such a rare use case, though, because you can't use Bun's globals.

> `bun install --yarn --no-save`

I think this combination of options was my missing piece of the puzzle. But the only way I can imagine this workflow locally is to commit `bunfig.toml` with those options, otherwise you'd forget, like you said.

But overall the only use case I can imagine is speeding things up on CI by using Bun there, and yarn locally in order to ensure compatibility and not accidentally introducing Bun-specific behavior.

**Reply by DieTapete** [reply]
i wonder why there is support for yarn lock files but not for npm's lock files.

**Reply by silvenon** [reply]
Yeah, not sure either, Bun chose yarn lockfile formatting to represent inspecting its lockfile for some reason (`bun bun.lockb`).
